lib/PrintAsObjC/PrintAsObjC.cpp

```diff
@@ -1612,28 +1614,34 @@ class ReferencedTypeFinder : private TypeVisitor<ReferencedTypeFinder> {
 
   /// Returns true if \p archetype has any constraints other than being
   /// class-bound ("conforms to" AnyObject).
-  static bool isConstrainedArchetype(const ArchetypeType *archetype) {
-    if (archetype->getSuperclass())
+  static bool isConstrained(ModuleDecl &mod,
+                            GenericSignature *sig,
+                            GenericTypeParamType *paramTy) {
+    if (sig->getSuperclassBound(paramTy, mod))
```

Comment:
@DougGregor recently added these methods to GenericSignature for doing archetype-like queries against dependent types. The module is needed in case we want to perform conformance lookups. I believe this is a consequence of same type constraints. Eg, if I constrain a dependent type X to be the same as a concrete type, and then ask the signature if a _member_ type Y of X is class-constrained or not, it has to perform the conformance lookup to get the type witness for Y.X. Or something.

So yeah, right now conformance lookups are global, and @DougGregor and I talked about how it might make more sense to always use the protocol's module as the module for lookups; since answering these queries differently depending on what module you're looking "from" doesn't make much sense either. But for now, we have to pass _something_ in, and passing in the module we're printing seems the most logical.